Source/Particles/Deposition/CurrentDeposition.H

```diff
@@ -1533,6 +1533,670 @@ void doChargeConservingDepositionShapeNImplicit (const amrex::ParticleReal * con
+                        // Throughout the following loop, xy_ takes the value e^{i m theta_}
+                        for (int imode=1 ; imode < n_rz_azimuthal_modes ; imode++) {
+                            // The factor 2 comes from the normalization of the modes
+                            const Complex djy_cmplx = 2._rt*this_Jy*xy_mid;
+                            amrex::Gpu::Atomic::AddNoRet( &Jy_arr(lo.x+i0_node+i, lo.y+k0_node+k, 0, 2*imode-1), djy_cmplx.real());
+                            amrex::Gpu::Atomic::AddNoRet( &Jy_arr(lo.x+i0_node+i, lo.y+k0_node+k, 0, 2*imode), djy_cmplx.imag());
+                            xy_mid = xy_mid*xy_mid0;
+                        }
+#endif
+                    }
+                }
+
+                // deposit Jz for this segment
+                amrex::Real this_Jz;
+                for (int i=0; i<=depos_order; i++) {
+                    for (int k=0; k<=depos_order-1; k++) {
+                        this_Jz = wqz*sz_cell[k]*(sx_old_node[i] + sx_new_node[i])/2.0_rt*seg_factor_z;
+                        amrex::Gpu::Atomic::AddNoRet( &Jz_arr(lo.x+i0_node+i, lo.y+k0_cell+k, 0, 0), this_Jz);
+#if defined(WARPX_DIM_RZ)
+                        Complex xy_mid = xy_mid0; // Throughout the following loop, xy_mid takes the value e^{i m theta}
+                        for (int imode=1 ; imode < n_rz_azimuthal_modes ; imode++) {
+                            // The factor 2 comes from the normalization of the modes
+                            const Complex djz_cmplx = 2._rt*this_Jz*xy_mid;
+                            amrex::Gpu::Atomic::AddNoRet( &Jz_arr(lo.x+i0_node+i, lo.y+k0_cell+k, 0, 2*imode-1), djz_cmplx.real());
+                            amrex::Gpu::Atomic::AddNoRet( &Jz_arr(lo.x+i0_node+i, lo.y+k0_cell+k, 0, 2*imode), djz_cmplx.imag());
+                            xy_mid = xy_mid*xy_mid0;
+                        }
+#endif
+                    }
+                }
+
+                // update old segment values
+                if (ns < num_segments-1) {
+                    x0_old = x0_new;
+                    z0_old = z0_new;
+                }
+
+            } // end loop over segments
+
+#elif defined(WARPX_DIM_1D_Z)
+
+            // compute cell crossings in Z-direction
+            const auto k_old = static_cast<int>(z_old-shift);
+            const auto k_new = static_cast<int>(z_new-shift);
+            int cell_crossings_z = std::abs(k_new-k_old);
+            num_segments += cell_crossings_z;
+
+            // need to assert that the number of cell crossings in each direction
+            // is within the range permitted by the number of guard cells
+            // e.g., if (num_segments > 2) ...
+
+            // compute dzp and the initial cell location used to find the cell crossings.
+            double const dzp = z_new - z_old;
+            const auto dirZ_sign = static_cast<int>(dzp < 0. ? -1 : 1);
+            double Zcell = k_old + shift + 0.5_rt*(1-dirZ_sign);
```

Comment:
as above, here I would explicitly cast the int to avoid surprises:
```suggestion
            double Zcell = static_cast<double>(k_old) + shift + 0.5*(1.0-dirZ_sign);
```

also here: consistently Real/ParticleReal or double?

Reply:
Will do.